src/Servers/IIS/IIS/src/Core/IISHttpContext.IO.cs

```diff
@@ -96,12 +98,19 @@ private async Task ReadBody()
 
                     var read = await AsyncIO.ReadAsync(memory);
 
+                    _consumedBytes += read;
```

Comment:
This needs to go inside the `if (read != -1)` check, and then MaxRequestBodySize needs to move down.

Reply:
Nice catch. Is there anyway we could create a regression test for this?

Reply:
IMO too hard, it would require me figuring out a way to consistently make IIS return an exception on every read. I don't know a way to make that happen consistently.